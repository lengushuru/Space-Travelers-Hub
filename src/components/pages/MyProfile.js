import React from 'react';
import { useSelector } from 'react-redux';
import styles from '../../styles/MyProfile.module.css';

const MyProfile = () => {
  const rockets = useSelector((state) => state.rockets.rockets);
  const reservedRockets = rockets.filter((rocket) => rocket.reserved);
  const missions = useSelector((state) => state.mission001.missions);
  const joinedMission = missions.filter((mission) => mission.reserved);

  return (
    <div className={styles.mainContainer}>
      <div className={styles.missionContainer}>
        <h2>My Missions</h2>
        <ul className={styles.UL}>
          {joinedMission.length ? (
            joinedMission.map((mission) => (
              <li key={mission.id}>{mission.mission_name}</li>
            ))
          ) : (
            <>
              <li>
                It looks like you haven&apos;t Joined any Missions yet.
                Don&apos;t forget to check out the Missions page to make a reservation.
              </li>
            </>
          )}
        </ul>
      </div>
      <div className={styles.rocketContainer}>
        <h2>My Rockets</h2>
        <ul className={styles.UL}>
          {reservedRockets.length ? (
            reservedRockets.map((rocket) => (
              <li key={rocket.id}>{rocket.name}</li>
            ))
          ) : (
            <>
              <li>
                It looks like you haven&apos;t reserved any rockets yet.
                Don&apos;t forget to check out the Rockets page to make a reservation.
              </li>
            </>
          )}
        </ul>
      </div>
    </div>
  );
};

export default MyProfile;
