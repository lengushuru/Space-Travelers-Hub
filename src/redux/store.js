import { configureStore } from '@reduxjs/toolkit';
import mission001Reducer from './missions/missionsSlice';

const store = configureStore({
  reducer: {
    mission001: mission001Reducer,
  },
});

export default store;
