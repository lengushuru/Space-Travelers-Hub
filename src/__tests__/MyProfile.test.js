import React from 'react';
import { cleanup, render } from '@testing-library/react';
import { Provider } from 'react-redux';
import store from '../redux/store';
import MyProfile from '../components/pages/MyProfile';

afterEach(cleanup);

describe('MyProfile', () => {
  it('MyProfile to match Snapshot', () => {
    const myProfile = render(<Provider store={store}><MyProfile /></Provider>);
    expect(myProfile).toMatchSnapshot();
  });
});
