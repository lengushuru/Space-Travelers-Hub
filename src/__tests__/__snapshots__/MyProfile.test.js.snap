// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyProfile MyProfile to match Snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="mainContainer"
      >
        <div
          class="missionContainer"
        >
          <p>
            missions container
          </p>
        </div>
        <div
          class="rocketContainer"
        >
          <h2>
            My Rockets
          </h2>
          <ul
            class="rocketsUL"
          >
            <li>
              It looks like you haven't reserved any rockets yet. Don't forget to check out the Rockets page to make a reservation.
            </li>
          </ul>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="mainContainer"
    >
      <div
        class="missionContainer"
      >
        <p>
          missions container
        </p>
      </div>
      <div
        class="rocketContainer"
      >
        <h2>
          My Rockets
        </h2>
        <ul
          class="rocketsUL"
        >
          <li>
            It looks like you haven't reserved any rockets yet. Don't forget to check out the Rockets page to make a reservation.
          </li>
        </ul>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
